microcosm_pubsub/dispatcher.py

```diff
@@ -24,25 +25,46 @@ class SQSMessageDispatcher:
     Dispatch batches of SQSMessages to handler functions.
 
     """
+    logger: Logger
+
     def __init__(self, graph):
         self.opaque = graph.opaque
         self.sqs_consumer = graph.sqs_consumer
         self.sqs_message_context = graph.sqs_message_context
         self.sqs_message_handler_registry = graph.sqs_message_handler_registry
         self.send_metrics = graph.pubsub_send_metrics
+        self.send_batch_metrics = graph.pubsub_send_batch_metrics
         self.max_processing_attempts = graph.config.sqs_message_dispatcher.message_max_processing_attempts
 
     def handle_batch(self, bound_handlers) -> List[MessageHandlingResult]:
         """
         Send a batch of messages to a function.
 
         """
-        instances = [
-            self.handle_message(message, bound_handlers)
-            for message in self.sqs_consumer.consume()
-        ]
+        with elapsed_time(self.opaque):
```

Comment:
I don't think we want to put this logging in `self.opaque`. This is similar to what I was doing on `request-instances`. If you put it in `self.opaque`, it will pick up a mysterious `elapsed_time` key which has no meaning once it leaves this service.

Reply:
Right. I'll make another object for this.